### ISSUE TITLE
[2015 CVPR Oral] Going Deeper with Convolutions

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#20]
해당 논문은 GoogLeNet으로도 불리며 Inception v1으로도 불림

Deep neural network의 성능을 올릴 수 있는 방법은 size를 늘리는 것인데, 이는 depth를 늘리거나 width를 늘리는 2가지 방법이 있음

하지만 size를 늘리면 parameter의 수가 많아져 overfitting에 취약하다는 단점과 computational resource가 증가한다는 단점이 있음

이러한 문제를 해결하기 위해 저자들은 sparsely connected architecture를 통해 해결함
(both issues would be by ultimately moving from fully connected to sparsely connected architectures)

## Model

![image](https://user-images.githubusercontent.com/49643709/119222693-93f40800-bb30-11eb-9604-cd74039635b0.png)

Sparsely connected architecture를 위해 저자들은 Inception architecture을 제안함

5 * 5, 3 * 3 conv를 그냥 사용하면 parameter가 많아 computation이 커지기에 1 * 1 conv로 reduction하여 사용

## Train

![image](https://user-images.githubusercontent.com/49643709/119222801-1381d700-bb31-11eb-9f71-39ee1e5c0e8e.png)
![image](https://user-images.githubusercontent.com/49643709/119222823-25637a00-bb31-11eb-90d7-2813ea6c5b81.png)
![image](https://user-images.githubusercontent.com/49643709/119223063-437daa00-bb32-11eb-8ac2-7c435dbd2dff.png)

그리고 AlexNet을 보면 충분히 network가 shallow해도 학습을 잘 한다는 것을 알 수 있기에, 중간 layer에 auxiliary classifier를 사용하여 더욱 학습을 잘하게 만듬

위 학습 방법을 통해 gradient signal을 증폭하고, 추가적으로 regularization 효과도 준다고 함

Auxiliary classifier로 얻은 loss는 0.3을 곱한 후 최종 loss로 업데이트함

학습은 SGD with momentum 0.9, fixed lr scheduler (decrease lr by 4% every 8 epochs)를 사용

Data augmentation으로는 random scale 후 crop을 함

## Inference
![image](https://user-images.githubusercontent.com/49643709/119223196-01089d00-bb33-11eb-97dd-6aaa15aac561.png)

Inference 시에는 auxiliary classifier를 모두 제거하고 Polyak averaing함
(Polyak averaing : model weight averaging)

그리고 최종 모델의 경우, ensemble + multi-crop을 이용

Multi-crop은 image를 길이가 짧은 쪽이 256, 288, 320, 352가 되게 resize한 후 가로가 길면 left, center, right 세로가 길면 top, center, bottom으로 정사각형 crop (256 * 256 , 288 * 288, 320 * 320, 352 * 352)함

해당 crop에서 4 corner + center 224 * 224 crop, 그리고 224 * 224 resize를 해줌

즉 4 (256, 288, 320, 352) * 3 (left, center, right or top, center, bottom) * 6 (4 corner, center, resize) * 2 (horizontal flip)으로 1장의 이미지에 대하여 144 crop을 가지고 classification함

## Detection

![image](https://user-images.githubusercontent.com/49643709/119223205-0c5bc880-bb33-11eb-9a52-58233bf86d66.png)

Detection의 경우는 R-CNN과 비슷함

Bounding box proposal에 selective search뿐만 아니라 multi-box prediction을 추가하며, classifier를 ensemble한 GoogLeNet을 사용함

시간이 부족하여 bounding box regression을 사용하지 않았다고 함
